pallets/staking/src/lib.rs

```
@@ -1951,24 +1915,23 @@ decl_module! {
             let stash = &ledger.stash;
 
             // Polymesh-Note - Make sure stash has valid permissioned identity.
-            if let Some(id) = <Identity<T>>::get_identity(stash) {
-                // Ensure stash's identity is be permissioned.
-                let mut id_pref = Self::permissioned_identity(id)
-                    .ok_or_else(|| Error::<T>::StashIdentityNotPermissioned)?;
-                ensure!(ledger.active >= <MinimumBondThreshold<T>>::get(), Error::<T>::InsufficientValue);
-                // Ensures that the passed commission is within the cap.
-                ensure!(prefs.commission <= Self::validator_commission_cap(), Error::<T>::InvalidValidatorCommission);
-                // Updates the running count.
-                if !<Validators<T>>::contains_key(stash) {
-                    // Ensure identity doesn't run more validators than the intended count.
-                    ensure!(id_pref.running_count < id_pref.intended_count, Error::<T>::HitIntendedValidatorCount);
-                    id_pref.running_count += 1;
-                }
-                PermissionedIdentity::insert(id, id_pref);
-
-                <Nominators<T>>::remove(stash);
-                <Validators<T>>::insert(stash, prefs);
+            // -----------------------------------------------------------------
+            let id = <Identity<T>>::get_identity(stash).ok_or(Error::<T>::StashIdentityDoesNotExist)?;
+            let mut id_pref = Self::permissioned_identity(id).ok_or(Error::<T>::StashIdentityNotPermissioned)?;
```

Semantic changes. We now error instead of silently accepting.